pkg/runtime/kubernetes/util.go

```diff
@@ -0,0 +1,104 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package kubernetes
+
+import (
+	"github.com/cockroachdb/errors"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/tensorchord/envd-server/api/types"
+	"github.com/tensorchord/envd-server/pkg/consts"
+	"github.com/tensorchord/envd-server/pkg/util"
+	"github.com/tensorchord/envd-server/pkg/util/imageutil"
+)
+
+func extractResourceRequest(env types.Environment) (v1.ResourceList, error) {
+	res := env.Resources
+	resRequest := v1.ResourceList{}
+	if res.CPU != "" {
+		cpu, err := resource.ParseQuantity(res.CPU)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse cpu resource")
+		}
+		resRequest[v1.ResourceCPU] = cpu
+	}
+	if res.Memory != "" {
+		mem, err := resource.ParseQuantity(res.Memory)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse memory resource")
+		}
+		resRequest[v1.ResourceMemory] = mem
+	}
+	if res.GPU != "" {
+		gpu, err := resource.ParseQuantity(res.GPU)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse gpu resource")
+		}
+		resRequest["nvidia/gpu"] = gpu
```

Comment:
How about creating a const.go to store `nvidia/gpu`.

Reply:
Could you please explain more about it?

Reply:
Just like `PodLabelUID`, maybe useful in other file.

Reply:
Do you mean to make a variable to store the string?

Reply:
right.